libs/SmartStore/src/com/salesforce/androidsdk/smartstore/store/AlterSoupLongOperation.java

```diff
@@ -220,7 +220,11 @@ private void alterSoupInternal(AlterSoupStep toStep)  {
 				reIndexSoup();
 			if (toStep == AlterSoupStep.RE_INDEX_SOUP) break;
 		case RE_INDEX_SOUP:
-			dropOldTable();		
+			dropOldTable();
+			// After upgrading to 4.5.5
+			// Functional indices are only picked after one query is run against the database
+			// See @W-14397675
+			store.getLongOperations();
 			if (toStep == AlterSoupStep.DROP_OLD_TABLE) break;
```

Comment:
3 tests were failing:

- testUpsertAgainstChangedSoup
- testDeleteAgainstChangedSoup
- testExactQueryAgainstChangedSoup

In those tests, we alter a soup (change its indexes) then run a query with explain query plan to make sure the new index is being used.

The test fails when the new index is a json functional index (e.g. json_extract (soup, '$.key')).

1) I verified that the indexes are correctly created (with additional logging and by querying the sqlite_master table).
2) I verified that json functional indices are being leveraged if setup initially (by doing explain query plan during other tests and with MobileSyncExplorer).
3) If I add a call to checkDatabaseIndexes() in the test after the alterSoup() call, then it works. All that checkDatabaseIndexes() does is calling dbOpenHelper.getWritableDatabase() followed by a query against the sqlite_master table to get the DDL statements that created the indices on the table. But debugging it looks like both the alterSoup and the checkDatabaseIndexes are going through the same SQLDatabase object.
4) If I simply run any query after alterSoup() e.g. store.getLongOperations(), then it works.

The "fix" for now is to run a query at the end of the alter soup operation.
Not sure why this is needed to "settle" the database.